### ISSUE TITLE
unique: move to module, add option to specify whether URIs open in current/new window

### ISSUE THREAD

**Issue by aidanholm** [luakit/luakit#395]
cc @xkpd3, @DanielFGray

**Comment by aidanholm** [comment]
Snipped IRC discussion:
```
14:57 <sdk> aidanh, firefox does this nicely there you have --url (existing window) --window (new window) --search (searches for whatever is passed in the default search)
15:01 <aidanh> Trouble is, luakit's command-line flags are controlled by C code; but luakit's unique instance support is controlled by Lua
15:01 <aidanh> Could move unique instance support into the luakit core, and remove the unique APIs
15:02 <aidanh> Could also add some API for rc.lua to parse extra command-line options
15:04 <aidanh> I don't particularly want to move stuff already done in Lua to C, unless there's a very good reason
15:05 <aidanh> And I'm not sure if I want to expose command-line flag parsing to Lua; they wouldn't show up in luakit --help, for example
15:06 <aidanh> Doing anything with the unique stuff is complicated anyway, because it's in the config file
15:06 <sdk> Well, it for me a config option would be absolutely fine
15:07 <sdk> just to statically control if luakit <url> should open a new window or try to open the url in an existing intance
15:07 <sdk> in globals.lua or so
```

**Comment by aidanholm** [comment]
FYI I'm also planning to eventually move the unique stuff from `rc.lua` to a Lua module, which would be a good time to add config options to that module. Command-line flags are still tricky, but if @DanielFGray is satisfied with a config option for now, we'll start with that. It'd look something like this:
```lua
local unique = require "unique"
unique.open_links_in_new_window = true
```

**Comment by aidanholm** [comment]
Cool; should be relatively simple to do; the only tricky bit is that the `unique` name is currently taken by the `unique` library. Perhaps that should be moved to `luakit.unique`; not sure yet.

**Comment by aidanholm** [comment]
Plan of action:

- Move `unique` library to `luakit.unique`
- `unique` will be a proxy for `luakit.unique` that prints deprecation warnings, to detect outdated `rc.lua`.
- Move unique code in `rc.lua` to `lib/unique_instance.lua`

Should be done by today

**Comment by aidanholm** [comment]
Merged at ab9b4a9e; see <luakit://help/doc/modules/unique_instance.html>
